src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1517,30 +1517,43 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
 
         assert((simdSize == 8) || (simdSize == 12) || (simdSize == 16) || (simdSize == 32));
 
-        if ((argCnt == 1) ||
-            ((vecCns.i64[0] == vecCns.i64[1]) && ((simdSize <= 16) || (vecCns.i64[2] == vecCns.i64[3]))))
-        {
-            // If we are a single constant or if all parts are the same, we might be able to optimize
-            // this even further for certain values, such as Zero or AllBitsSet.
-
-            if (vecCns.i64[0] == 0)
+        bool allBitsAreUnset        = true;
+        bool allBitsAreSet          = true;
+        int  i32ComponentsToInspect = argCnt == 1 ? 2 : simdSize / 4;
```

Comment:
Should we assert that `simdSize % 4 == 0`?

Reply:
yeah sure, I was about to comment on your feedback but wanted to fix the arm64 issue first which is unrelated to my changes but fails on CI on my new tests.